### PR TITLE
Improve skymatch documentation

### PR DESCRIPTION
**Pull request by mcara**
This PR adresses some of the issues described in https://github.com/STScI-JWST/jwst/issues/741:

* Fixed `astrodrizzle` link and spacing in issue 1.
* Expanded explanation of who algorithm works (what it does) based on info that I got on JIRA after I requested clarification on what issue 2 is about.
* Issue 3.b - fixed "boldness" of `subtract` parameter.

Issues 3.a and 4 are invalid.

### CODE REVIEW

**Comment by mcara** [comment]
@hbushouse Please verify the language of expanded description of the overlap regions and what a background is - see https://github.com/STScI-JWST/jwst/pull/2084/files#diff-18dd9bdf16f9a89f099788aec9c4ebb1R17 and https://github.com/STScI-JWST/jwst/pull/2084/files#diff-18dd9bdf16f9a89f099788aec9c4ebb1R70

File: docs/jwst/skymatch/README.rst [hunk 1]
```diff
@@ -14,8 +14,10 @@ cf. :doc:`mrs_imatch step <../mrs_imatch/README>`) of a set of input images
 and computes the signal offsets for each image that will minimize
 the residuals across the entire set in the least squares sence. This comparison
 is performed directly on input images without resampling them onto a common
-grid. By default the sky value computed for each image is recorded, but
-not actually subtracted from the images.
+grid. The overlap regions are computed directly on the sky (celestial sphere)
+for each pair of input images. By default the sky value computed for each image
+is recorded, but not actually subtracted from the images. Also note that the
+meaning of "sky background" is depends on the chosen sky computation method.
```
Comment (hbushouse) [inline docs/jwst/skymatch/README.rst]: Remove the word "is" (should be 'meaning of "sky background" depends on the chosen ...')

File: docs/jwst/skymatch/README.rst [hunk 2]
```diff
@@ -65,6 +67,21 @@ two other methods have been introduced: ``'globalmin'`` and
   algorithm to find a baseline sky value common to all input images
   and the ``'match'`` algorithm to "equalize" sky values among images.
 
+In methods that find sky background levels in each image (``'localmin'``) or
+a single level for all images (``'globalmin'``), usually statistics is
```
Comment (hbushouse) [inline docs/jwst/skymatch/README.rst]: ... usually statistics **are** ...

File: docs/jwst/skymatch/README.rst [hunk 3]
```diff
@@ -65,6 +67,21 @@ two other methods have been introduced: ``'globalmin'`` and
...
+computed using sigma-clipping. If input images contain vast swaths of empty
+sky, then the sigma-clipping algorithm should be able to automatically
+exclude (clip) contributions from bright compact sources.
+In this case the measured "sky background" is the
+measured signal level from the "empty sky". On the other hand, the
+``'match'`` method compares *total* signal levels integrated over those regions
+in the images that correspond to common ("overlap") regions on the celestial
+sphere for both images being compared (comparison is pair-wise).
+Often, this method is used when there are no large
```
Comment (hbushouse) [inline docs/jwst/skymatch/README.rst]: Slight rewording to "This method is often used when ..."

File: docs/jwst/skymatch/README.rst [hunk 4]
```diff
@@ -65,6 +67,21 @@ two other methods have been introduced: ``'globalmin'`` and
...
+"empty sky" regions in the images such as when a large nebula occupies most
+part of the view. This method cannot measure "true background" but
```
Comment (hbushouse) [inline docs/jwst/skymatch/README.rst]: Need to add "or" in the middle of "... large nebula occupies most **or** part of the view."

File: docs/jwst/skymatch/README.rst [hunk 5]
```diff
@@ -83,11 +100,11 @@ The ``skymatch`` step has the following optional arguments:
...
 
   .. note::
-     This setting applies *only* when `skymethod` parameter is
-     either `'match'` or `'global+match'`.
+    This setting applies *only* when ``skymethod`` parameter is
+    either ``'match'`` or ``'global+match'``.
 
 * ``subtract``: A boolean indicating whether computed sky background values
-    be subtracted from image data. (Default = `False`)
+  be subtracted from image data. (Default = `False`)
```
Comment (hbushouse) [inline docs/jwst/skymatch/README.rst]: "... **are to** be subtracted from image data."

**Comment by mcara** [comment]
@hbushouse please check this last commit.

**Pull request merged**